src/plugins/directoryvalue/log.hpp

```diff
@@ -27,10 +27,12 @@ void logKeySet (kdb::KeySet const & keys)
 	for (auto key : keys)
 	{
 		std::string metadata;
-		key.rewindMeta ();
-		while (kdb::Key meta = key.nextMeta ())
+		ckdb::KeySet * metaKeys = ckdb::keyMeta (key.getKey ());
+
+		for (elektraCursor it = 0; it < ckdb::ksGetSize (metaKeys); ++it)
 		{
-			metadata += ", “" + meta.getName () + "”: “" + meta.getString () + "”";
+			const kdb::Key curMeta (ckdb::ksAtCursor (metaKeys, it));
+			metadata += ", “" + curMeta.getName () + "”: “" + curMeta.getString () + "”";
```

Comment:
This has different behaviour than the old version, so the failing test if this function is used could be explained. `keyMeta(k)` will create the metadata `KeySet` of `k`, if `k` doesn't have one. But `keyNextMeta(k)` simply returned `NULL`, without touching `k`.

But I don't think this should change anything, since with the new API there is no way of knowing whether a `k` has a metadata `KeySet` or not. At least not without accessing the `struct _Key` directly...